api/processor_api.py

```diff
@@ -191,157 +91,20 @@ def map_config(config, options):
-                    config_dict[f'Area_{index}'] = config_dict[area_name]
-                    config_dict.pop(area_name)
-            return config_dict
-
-        def reestructure_cameras(config_dict):
-            """Ensure that all [Source_0, Source_1, ...] are consecutive"""
-            source_names = [x for x in config_dict.keys() if x.startswith("Source")]
-            source_names.sort()
-            for index, source_name in enumerate(source_names):
-                if f'Source_{index}' != source_name:
-                    config_dict[f'Source_{index}'] = config_dict[source_name]
-                    config_dict.pop(source_name)
-            return config_dict
-
-        def delete_camera_from_areas(camera_id, config_dict):
-            areas = {key: config_dict[key] for key in config_dict.keys() if key.startswith("Area")}
-            for key, area in areas.items():
-                cameras = area['Cameras'].split(',')
-                if camera_id in cameras:
-                    cameras.remove(camera_id)
-                    if len(cameras) == 0:
-                        logger.warning(f'After removing the camera "{camera_id}", the area "{area["Id"]} - {area["Name"]}" was left with no cameras and deleted')
-                        config_dict.pop(key)
-                    else:
-                        config_dict[key]['Cameras'] = ",".join(cameras)
-
-            config_dict = reestructure_areas(config_dict)
-
-            return config_dict
-
-        def get_cameras(options):
-            config = extract_config(config_type='cameras')
-            return [map_camera(x, config, options) for x in config.keys()]
-
-        def update_and_restart_config(config_dict):
-            update_config_file(config_dict)
-
-            # TODO: Restart only when necessary, and only the threads that are necessary (for instance to load a new video)
-            success = restart_processor()
-            return success
-
         @app.get("/process-video-cfg")
         async def process_video_cfg():
             logger.info("process-video-cfg requests on api")
-            self._cmd_queue.put(Commands.PROCESS_VIDEO_CFG)
+            self.queue_manager.cmd_queue.put(Commands.PROCESS_VIDEO_CFG)
             logger.info("waiting for core's response...")
-            result = self._result_queue.get()
+            result = self.queue_manager.result_queue.get()
             return result
 
         @app.get("/stop-process-video")
         async def stop_process_video():
             logger.info("stop-process-video requests on api")
-            self._cmd_queue.put(Commands.STOP_PROCESS_VIDEO)
+            self.queue_manager.cmd_queue.put(Commands.STOP_PROCESS_VIDEO)
             logger.info("waiting for core's response...")
-            result = self._result_queue.get()
+            result = self.queue_manager.result_queue.get()
             return result
 
         @app.get("/config", response_model=ConfigDTO)
```

Comment:
What do you think about creating a `config` subapi?

Reply:
Yes, make sense move these endpoints to an independet file.
I did it in the commit 7395edded386ab1239358207128a5f26fe7f612e.